pkg/kubelet/kubelet.go

```diff
@@ -1386,7 +1386,11 @@ func (kl *Kubelet) GetKubeletContainerLogs(podFullName, containerName, tail stri
 		// waiting state.
 		return err
 	}
-	return kl.containerManager.GetContainerLogs(containerID, tail, follow, stdout, stderr)
+	pod, ok := kl.GetPodByFullName(podFullName)
```

Comment:
IMO, GetContainerLogs should simply find the containerID itself and return error if it's not found. Also, not relevant to this PR, but why can't GetContainerLogs take containerName as an argument -- this seems to fit well with our pod-level abstraction.